File: FilmHive/filmHive.Model/FilmFavorite.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace filmHive.Model
{
    public class FilmFavorite
    {
        public int FilmFavoriteId { get; set; }

        public int MovieId { get; set; }

        public int UserId { get; set; }

        public DateTime? AddedAt { get; set; }

        public bool IsDeleted { get; set; }

        public bool IsActive { get; set; }

        public DateTime? TimeOfDeletion { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime? ModifiedAt { get; set; }

        public int? ModifiedBy { get; set; }

        public virtual Film? Movie { get; set; }

        public virtual User? User { get; set; }
    }
}

File: FilmHive/UI/filmHive/models/filmFavorite.js
import Film from "./film";
import User from "./user";

class FilmFavorite{

    constructor(data){
        this.filmFavoriteId = data.filmFavoriteId;
        this.movieId = data.movieId;
        this.userId = data.userId;
        this.addedAt = data.addedAt;
        this.isDeleted = data.isDeleted;
        this.isActive = data.isActive;
        this.timeOfDeletion = data.timeOfDeletion;
        this.createdAt = data.createdAt;
        this.modifiedAt = data.modifiedAt;
        this.modifiedBy = data.modifiedBy;
        this.movie= data.movie?new Film(data.movie):null;
        this.user = data.user ? new User(data.user) : null;
    }
}
export default FilmFavorite;